packages/java/engine-core/src/main/java/com/vaadin/hilla/engine/AotEndpointProvider.java

```diff
@@ -0,0 +1,139 @@
+    private static final String SPRING_BOOT_APPLICATION_CLASS_NAME = "org.springframework.boot.autoconfigure.SpringBootApplication";
+    private final EngineConfiguration engineConfiguration;
+
+    public AotEndpointProvider(EngineConfiguration engineConfiguration) {
+        this.engineConfiguration = engineConfiguration;
+    }
+
+    public List<Class<?>> findEndpointClasses()
+            throws IOException, InterruptedException {
+        // Prepares all variables based on the provided configuration
+        var aotOutput = engineConfiguration.getBuildDir()
+                .resolve("spring-aot/main");
+        var classesDirectory = aotOutput.resolve("classes");
+        var applicationClass = (engineConfiguration.getMainClass() != null)
+                ? engineConfiguration.getMainClass()
+                : MainClassFinder.findSingleMainClass(
+                        engineConfiguration.getClassesDir().toFile(),
+                        SPRING_BOOT_APPLICATION_CLASS_NAME);
+
+        if (applicationClass == null) {
+            LOGGER.warn("This project has not been recognized as a Spring Boot"
+                    + " application because a main class could not be found."
+                    + " Hilla services will not be available.");
+            return List.of();
+        }
+
+        var classpath = engineConfiguration.getClasspath().stream()
+                .filter(Files::exists).toList();
+        var settings = Stream.of("-cp",
+                classpath.stream().map(AotEndpointProvider::quotePath)
+                        .collect(Collectors.joining(File.pathSeparator)),
+                "org.springframework.boot.SpringApplicationAotProcessor",
+                applicationClass, quotePath(aotOutput.resolve("sources")),
+                quotePath(aotOutput.resolve("resources")),
+                quotePath(classesDirectory), engineConfiguration.getGroupId(),
+                engineConfiguration.getArtifactId()).toList();
+        var argsFile = engineConfiguration.getBuildDir()
+                .resolve("aot-args-" + System.currentTimeMillis() + ".txt");
+        Files.write(argsFile, settings);
+        var javaExecutable = ProcessHandle.current().info().command()
+                .orElse(Path.of(System.getProperty("java.home"), "bin", "java")
+                        .toString());
+
+        // Runs the SpringApplicationAotProcessor to generate the
+        // reflect-config.json file. This comes from the `process-aot` goal.
+        new ProcessBuilder().inheritIO().command(javaExecutable, "@" + argsFile)
+                .start().waitFor();
+        Files.delete(argsFile);
+
+        var json = aotOutput.resolve(Path.of("resources", "META-INF",
+                "native-image", engineConfiguration.getGroupId(),
+                engineConfiguration.getArtifactId(), "reflect-config.json"));
+
+        if (!Files.isRegularFile(json)) {
+            throw new ParserException("Aot file reflect-config.json not found");
```

Comment:
It's checking that it exists and it's a file, no check for corrupted information is done here.

Reply:
In what circumstances do we expect this to happen? In case it happens, should the users just ignore it, and retry to build? And could we include more info in the exception message, so that if it was a recurrent exception, users can have easier times finding what's the issue? Without some context info, user might get confused about why they're getting an exception regarding the AOT, while they were just trying to build their app, without having some intentions of doing a native build.

Reply:
I propose to modify the error message to honestly admit that we have run the Spring AOT tool and it hasn't generate the desired file for some reason. Normally, it has printed out something in the console just before by itself if something was wrong.

Reply:
Sounds good!